creator-node/src/snapbackSM.js

```diff
@@ -0,0 +1,110 @@
+const config = require('./config')
+const Bull = require('bull')
+const utils = require('./utils')
+const { logger } = require('./logging')
+
+const DevDelayInMS = 10000
+
+/*
+  Snap back state machine
+  Ensures file availability through sync and user replica operations
+*/
+class SnapbackSM {
+  constructor (audiusLibs) {
+    this.audiusLibs = audiusLibs
+    this.initialized = false
+    // Toggle to switch logs
+    this.debug = true
+    // Throw an error if running as creator node and no libs are provided
+    if (!this.audiusLibs && !config.get('isUserMetadataNode')) {
+      throw new Error('Invalid libs provided to SnapbackSM')
```

Comment:
@hareeshnagaraj where is this error handled? i dont think ServiceRegistry initServices has a try/catch

Reply:
this error should crash the node - this means libs failed to initialize while running as a creator node. is a different behavior expected?